database/sinkdb/sinkdb.go

```diff
@@ -53,9 +53,8 @@ func (db *DB) Get(ctx context.Context, key string, v proto.Message) (found bool,
 	if err != nil {
 		return false, err
 	}
-	buf := db.state.get(key)
-	// TODO(jackson): propagate real key existence bool
-	if len(buf) == 0 {
+	buf, found := db.state.get(key)
+	if !found {
```

Comment:
I think it would be better to call Unmarshal unconditionally:

```go
buf, found := db.state.Get(key)
return found, proto.Unmarshal(buf, v)
```

The reason is that sometimes the caller doesn't care if the key doesn't exist — they might want to ignore the found flag and see a zero value as if that had been explicitly stored. Separately, sometimes the caller might want to reuse a protobuf struct to avoid allocating. If both of these happen at the same time, the right thing to do is to `v.Reset()` so the caller gets a zero value out. Unmarshal does this; it calls Reset before unmarshaling, even if the byte slice is empty.